python/lsst/db/engineFactory.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+
+# LSST Data Management System
+# Copyright 2008-2015 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+This module exposes engine from SQLAlchemy.
+
+@author  Jacek Becla, SLAC
+"""
+
+# standard library imports
+from ConfigParser import ConfigParser
+import logging as log
+import os
+
+# related third-package library imports
+import sqlalchemy
+from sqlalchemy.engine.url import URL
+
+
+####################################################################################
+
+def getEngineFromFile(fileName):
+    """
+    Initializes and returns SQLAlchemy engine using values provided through the
+    file. The file must contain "database" section with key "url" defined.
+    Optionally it can contain **kwargs that can be passed to the SQLAlchemy
+    create_engine(). For  details see
+    http://docs.sqlalchemy.org/en/rel_1_0/core/engines.html
+
+    Note, if mysql sees "localhost" it switches to using socket, even if port is
+    specified. Commonly used way around it is to specify "127.0.0.1" as port for
+    local access.
+
+    Example file:
+
+    [database]
+    url = mysql+mysqldb://joe:myPassword@localhost:3306/?unix_socket=/tmp/mysql.sock
+    echo = yes
+    pool_size = 5
+    """
+    if fileName.startswith('~'):
+        fileName = os.path.expanduser(fileName)
+    if not os.path.isfile(fileName):
+        raise IOError
+    parser = ConfigParser()
+    parser.read(fileName)
+    options = dict(parser.items("database"))
```

Comment:
If there is no [database] section then `parser.items()` will throw. I'm OK with just passing that exception to caller, but if you want "better" diagnostics we can catch here and re-throw with more information. Currently it raises something like:

    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/u2/salnikov/STACK/Linux64/anaconda/2.1.0-4-g35ca374/lib/python2.7/ConfigParser.py", line 642, in items
        raise NoSectionError(section)
    ConfigParser.NoSectionError: No section: 'database'